spring-data-mongodb/src/main/java/org/springframework/data/mongodb/repository/support/SimpleReactiveMongoRepository.java

```diff
@@ -97,7 +98,7 @@
 						return Mono.error(new IncorrectResultSizeDataAccessException(1));
 					}
 					return Mono.just(vals.iterator().next());
-				}).single();
+				}).singleOrEmpty();
```

Comment:
We should use `.next()` here. We've already covered the case if there's more than one element.